sensor/common/signal/singleton.go

```diff
@@ -1,19 +1,17 @@
 package signal
 
 import (
-	v1 "github.com/stackrox/rox/generated/api/v1"
-	"github.com/stackrox/rox/sensor/common/message"
+	"github.com/stackrox/rox/generated/storage"
 )
 
-// New creates a new signal service
-func New(pipeline Pipeline, indicators chan *message.ExpiringMessage, opts ...Option) Service {
+const maxBufferSize = 10000
+
+func NewService(opts ...Option) Service {
 	srv := &serviceImpl{
-		queue:            make(chan *v1.Signal, maxBufferSize),
-		indicators:       indicators,
-		processPipeline:  pipeline,
+		queue:            make(chan *storage.ProcessSignal, maxBufferSize),
```

Comment:
Not 100% sure why we had this `queue: make(chan *v1.Signal, maxBufferSize)` before because I believe it was not used 🤷 but since now `queue: make(chan *storage.ProcessSignal, maxBufferSize)` is used, I'd add metrics and make is configurable with `ScaleSizeOnNonDefault` +  and env variable.

Reply:
Asking for both this and the `serivice_impl.go` comment, could you provide an example of the metrics you'd like added? I'm not 100% sure how to look for those in the sensor code.

Reply:
Sure! [Here's](https://github.com/stackrox/stackrox/blob/6a15c1e89a503cddf0681d4f906e2f9b854b85b6/sensor/common/sensor/buffered_stream.go#L15) how we do it for the `BufferedStream`.